core/services/keeper/orm_test.go

```diff
@@ -44,9 +48,11 @@ func newUpkeep(registry keeper.Registry, upkeepID int64) keeper.UpkeepRegistrati
 }
 
 func assertLastRunHeight(t *testing.T, db *gorm.DB, upkeep keeper.UpkeepRegistration, height int64) {
-	err := db.Find(&upkeep).Error
-	require.NoError(t, err)
-	require.Equal(t, height, upkeep.LastRunBlockHeight)
+	gomega.NewGomegaWithT(t).Eventually(func() int64 {
```

Comment:
Is this really necessary? `assertLastRunHeight` is called just after `SetLastRunHeightForUpkeepOnJob` which is synchronous. You might be running into issues with `t.Parallel`?

Reply:
@archseer not really as I understood. `assertLastRunHeight` is running after pipeline is finished, but not after `SetLastRunHeightForUpkeepOnJob` (https://github.com/smartcontractkit/chainlink/blob/develop/core/services/keeper/upkeep_executer.go#L217). So in my changes, `assertLastRunHeight` will run only after the last height is updated, otherwise, the test will be failed.

Reply:
`assertLastRunHeight` is being used in the `orm` tests and `upkeep_executer` tests. In the `orm` tests it is being used synchronously, but in the `upkeep_executer` it is not. 

Perhaps we should make two separate helper functions, since this can be confusing as to why the orm test would need to wait to assert.

Reply:
Created a separate function for non-orm tests

Reply:
Note: the pipeline runner does have a channel intended for testing now so that we're able to do asserts without relying on something like gomega: https://github.com/smartcontractkit/chainlink/blob/ea7dab1aae73bd129e9e506dcd5498ecc9fc3dba/core/services/pipeline/runner.go#L51

Reply:
This is how it's used elsewhere: https://github.com/smartcontractkit/chainlink/blob/5b07fa79eafad31dba2badd08b9b208c17f7dbce/core/services/vrf/delegate_test.go#L409-L414